parsing/pprintast.ml

```diff
@@ -407,6 +408,8 @@ and core_type1 ctxt f x =
          |_ ->
              pp f "@[<hov2>(module@ %a@ with@ %a)@]" longident_loc lid
                (list aux  ~sep:"@ and@ ")  cstrs)
+    | Ptyp_open(li, ct) ->
+       pp f "@[<hov2>%a.(%a)@]" longident_loc li (core_type ctxt) ct
     | Ptyp_extension e -> extension ctxt f e
     | _ -> paren true (core_type ctxt) f x
```

Comment:
To avoid this kind of bugs, I like to list all non-handled cases explicitly at the last level of the pretty-printing hierarchy (so on the `paren true ...` line here): new constructors added in the future become non-exhaustive-matching warnings, so we are reminded to update the code. So my suggestion would be to use

```ocaml
| (Ptyp_arrow _ | Ptyp_alias _ | Ptyp_poly _) -> paren true (core_type ctxt) f x
```

Reply:
Shall I add that?  Let me know and I'll do it posthaste. [it's a little outside the scope of a strict bugfix, or I'd just do it right now]

Reply:
I would be in favor as it is in the "making sure that we don't get bitten twice" mitigation category, but your choice.